src/screens/DriveScreen/index.tsx

```diff
@@ -21,23 +21,35 @@ import AppScreen from '../../components/AppScreen';
 import { ArrowDown, ArrowUp, CaretLeft, DotsThree, MagnifyingGlass, Rows, SquaresFour } from 'phosphor-react-native';
 import { asyncStorage } from '../../services/asyncStorage';
 import { NativeStackNavigationProp } from '@react-navigation/native-stack';
-import { DriveListType, SortDirection } from '../../types/drive';
+import { DriveListType, SortDirection, SortType } from '../../types/drive';
+import SortModal from '../../components/modals/SortModal';
+import fileService from '../../services/file';
 
 function DriveScreen(): JSX.Element {
   const navigation = useNavigation<NativeStackNavigationProp<any>>();
   const route = useRoute();
   const dispatch = useAppDispatch();
+  const [sortModalOpen, setSortModalOpen] = useState(false);
+  const [sortOptions, setSortOptions] = useState({
```

Comment:
To put some downside, this variable should be called sortMode for consistency with the type it meets and we avoid the plural for an object as well

Reply:
yep, that makes more sense